tests/RecordTests/FunctionalTest/k4a_cpp_ft.cpp

```diff
@@ -0,0 +1,453 @@
+
+        ASSERT_TRUE(color.get_size() > 0);
+        ASSERT_TRUE(color.get_format() == K4A_IMAGE_FORMAT_COLOR_MJPG);
+        ASSERT_TRUE(color.get_width_pixels() == 1920);
+        ASSERT_TRUE(color.get_height_pixels() == 1080);
+        ASSERT_TRUE(color.get_stride_bytes() == 0);
+        ASSERT_TRUE(color.get_device_timestamp() != std::chrono::microseconds(0));
+        ASSERT_TRUE(color.get_system_timestamp() != std::chrono::microseconds(0));
+        ASSERT_TRUE(color.get_exposure() != std::chrono::microseconds(0));
+        ASSERT_TRUE(color.get_white_balance() != 0);
+        ASSERT_TRUE(color.get_iso_speed() != 0);
+
+        // test that they don't fail
+        color.set_timestamp(std::chrono::microseconds(0x1234));
+        color.set_exposure_time(std::chrono::microseconds(500));
+        color.set_white_balance(500);
+        color.set_iso_speed(500);
+    }
+}
+
+TEST_F(k4a_cpp_ft, record)
+{
+    device kinect = device::open(0);
+    k4a_device_configuration_t config = K4A_DEVICE_CONFIG_INIT_DISABLE_ALL;
+    config.color_resolution = K4A_COLOR_RESOLUTION_1080P;
+    config.depth_mode = K4A_DEPTH_MODE_NFOV_UNBINNED;
+    config.synchronized_images_only = true;
+    kinect.start_cameras(&config);
+    kinect.start_imu();
+
+    record recorder = record::create("./k4a_cpp_ft.mkv", kinect, config);
+    {
+        record recorder2 = record::create("./k4a_cpp_ft_2.mkv", kinect, config);
+        ASSERT_TRUE(recorder != recorder2);
+        ASSERT_FALSE(recorder == recorder2);
+
+        record recorder4 = std::move(recorder2); // deep copy
+        ASSERT_TRUE(recorder4 != nullptr);
+        ASSERT_FALSE(recorder4 == nullptr);
+
+        recorder2.close();
+        recorder4.close();
+    }
+
+    recorder.add_tag("K4A_CPP_FT_ADD_TAG", "K4A_CPP_FT_ADD_TAG");
+    recorder.add_imu_track();
+
+    std::string k4a_cpp_ft_attachment = "K4A_CPP_FT_ADD_ATTACHMENT";
+    recorder.add_attachment(k4a_cpp_ft_attachment.c_str(),
+                            (uint8_t *)k4a_cpp_ft_attachment.data(),
+                            k4a_cpp_ft_attachment.size());
+
+    k4a_record_video_settings_t vid_settings = { 1920, 1080, 30 };
+    std::string k4a_cpp_ft_custom_vid_track = "K4A_CPP_FT_CUSTOM_VID_TRACK";
+    recorder.add_custom_video_track(k4a_cpp_ft_custom_vid_track.c_str(), "V_MPEG1", nullptr, 0, &vid_settings);
+
+    k4a_record_subtitle_settings_t st_track = { false };
+    std::string k4a_cpp_ft_custom_subtitle_track = "CUSTOM_K4A_SUBTITLE_TRACE";
+    recorder.add_custom_subtitle_track(k4a_cpp_ft_custom_subtitle_track.c_str(), "V_MPEG1", nullptr, 0, &st_track);
+
+    recorder.write_header();
+
+    for (int x = 0; x < 100; x++)
+    {
+        capture capture;
+        k4a_imu_sample_t imu;
+        if (kinect.get_capture(&capture, std::chrono::milliseconds(1000)))
+        {
+            recorder.write_capture(capture);
+        }
+        while (kinect.get_imu_sample(&imu, std::chrono::milliseconds(0)))
```

Comment:
>while [](start = 8, length = 5)

I think this can hang forever if the system somehow is running slow enough that it can't keep up with incoming IMU samples - we might want a max here so the test agent doesn't hang indefinitely if some other process on the test agent is hammering the processor